notebooks/example_create_agent.ipynb

```diff
@@ -0,0 +1,284 @@
+    "5. Now, you can copy the new token that you have generated and paste it into the script below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "import subprocess\n",
+    "import sys\n",
+    "import os\n",
+    "\n",
+    "def install_sdk_library():\n",
+    "    token = getpass.getpass(\"Input Your Personal Access Token: \")\n",
+    "\n",
+    "    cmd = f\"pip install -e git+https://{token}@github.com:GDP-ADMIN/gen-ai-internal.git#egg=gdplabs_gen_ai\"\n",
+    "\n",
+    "    with subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True) as process:\n",
+    "        stdout, stderr = process.communicate()\n",
+    "\n",
+    "        if process.returncode != 0:\n",
+    "            sys.stdout.write(stderr)\n",
+    "            raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)\n",
+    "        else:\n",
+    "            sys.stdout.write(stdout)\n",
+    "\n",
+    "install_sdk_library()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Add Environment Variable\n",
+    "\n",
+    "You will need two provide two API Keys. OpenAI API Key and SerpAPI API Key."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.environ[\"OPENAI_API_KEY\"] = getpass.getpass(\"OPENAI API KEY: \")\n",
+    "os.environ[\"SERPAPI_API_KEY\"] = getpass.getpass(\"SERPERAI API KEY: \")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create and execute Agent"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define Tools for the Agent\n",
+    "\n",
+    "In this section, we will create two tools that will be used for our agent determining the answer for the query/question given by user. \n",
+    "1. `Search Tool` - Utilize Google search API to get additional information\n",
+    "2. `Multiplier Tool` - Accept two numbers and return the result of their multiplication\n",
+    "\n",
+    "To define the tools, we use interface on Gen AI SDK. You can read more details on [Agent Tools](https://docs.glair.ai/generative-internal/modules/agent-orchestrator/tools)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from gdplabs_gen_ai.agent_orchestrator.agent_executor import initialize_agent_executor\n",
+    "from gdplabs_gen_ai.agent_orchestrator.agent_type import AgentType\n",
+    "from gdplabs_gen_ai.agent_orchestrator.tools import BaseTool\n",
+    "from gdplabs_gen_ai.inference_orchestrator.llm import OpenAILLM\n",
+    "from langchain.utilities import SerpAPIWrapper\n",
+    "\n",
```

Comment:
Received error when running this block

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[9], line 1
----> 1 from gdplabs_gen_ai.agent_orchestrator.agent_executor import initialize_agent_executor
      2 from gdplabs_gen_ai.agent_orchestrator.agent_type import AgentType
      3 from gdplabs_gen_ai.agent_orchestrator.tools import BaseTool

ModuleNotFoundError: No module named 'gdplabs_gen_ai'
```

Even though the installation was successful.

Reply:
This should be because the SDK code still not in branch main yet.

You can change it to for the time being
```python
cmd = f"pip install -e git+https://{token}@github.com/GDP-ADMIN/gen-ai-internal.git@agent-tool#egg=gdplabs_gen_ai"
```

Reply:
after change the url  to include branch name, **gdplabs_gen_ai** as library is successfully loaded 
However now receive this error:
```

     25 class BasePromptBuilder:
     26     """Base PromptBuilder abstract interface.
     27 
     28     Attributes:
   (...)
     36         _format_prompt_config(): Constructs multiple prompt template in a dictionary.
     37     """
---> 39     def __init__(self, template: str | Dict[str, str], prompt_keys: List[str] = None) -> None:
     40         """Initializes a new instance of the PromptBuilder class.
     41 
     42         Args:
     43             template (str): A string contains the prompt template.
     44             prompt_keys: List[str]: A list of key variables in the prompt. Defaults to ["question", "context"].
     45         """
     46         self.prompt_keys = ["question", "context"]

TypeError: unsupported operand type(s) for |: 'type' and '_GenericAlias'
```

I am using python 3.9

Reply:
The operator is only available for python 3.10 onward. Since Gen AI SDK python requirement is python 3.10 you need to use python 3.10. Should we put this info on the notebook or the info on Gen AI SDK page already sufficient?

Reply:
Ah I see, I was making assumption.
I think it is sufficient enough in SDK page